embedchain/llm/base.py

```diff
@@ -174,27 +174,37 @@ def query(self, input_query: str, contexts: List[str], config: BaseLlmConfig = N
         :return: The answer to the query or the dry run result
         :rtype: str
         """
-        query_config = config or self.config
-
-        if self.is_docs_site_instance:
-            query_config.template = DOCS_SITE_PROMPT_TEMPLATE
-            query_config.number_documents = 5
-        k = {}
-        if self.online:
-            k["web_search_result"] = self.access_search_and_get_results(input_query)
-        prompt = self.generate_prompt(input_query, contexts, **k)
-        logging.info(f"Prompt: {prompt}")
-
-        if dry_run:
-            return prompt
-
-        answer = self.get_answer_from_llm(prompt)
-
-        if isinstance(answer, str):
-            logging.info(f"Answer: {answer}")
-            return answer
-        else:
-            return self._stream_query_response(answer)
+        try:
```

Comment:
Not a great idea to put try/except around several lines of code. Can you please change this to make sure we only wrap the line where the exception is happened?